sonar-dotnet-shared-library/src/main/java/org/sonar/plugins/dotnet/tests/VisualStudioCoverageXmlReportParser.java

```diff
@@ -34,22 +34,27 @@
   private static final Logger LOG = Loggers.get(VisualStudioCoverageXmlReportParser.class);
   private final Predicate<String> isSupported;
 
-  public  VisualStudioCoverageXmlReportParser(Predicate<String> isSupported) {
+  VisualStudioCoverageXmlReportParser(Predicate<String> isSupported) {
     this.isSupported = isSupported;
   }
 
   @Override
   public void accept(File file, Coverage coverage) {
     LOG.debug("The current user dir is '{}'.", System.getProperty("user.dir"));
     LOG.info("Parsing the Visual Studio coverage XML report " + file.getAbsolutePath());
-    new Parser(file, coverage).parse();
+    Parser parser = new Parser(file, coverage);
+    parser.parse();
   }
 
   private class Parser {
 
     private final File file;
-    private final Map<Integer, List<Integer>> coveredLines = new HashMap<>();
-    private final Map<Integer, List<Integer>> uncoveredLines = new HashMap<>();
+    /**
+     * The outer map key is the file ID.
+     * The inner map key is the line number.
+     * The inner map values are a list of Boolean - true when the line is covered, false when it is not.
+     */
+    private final Map<Integer, Map<Integer, List<Boolean>>> coveragePerLinePerFile = new HashMap<>();
```

Comment:
An option would be to create a new class LineCoverage and have a Map<Integer, List<LineCoverage>>. I think this will make the code a bit more easy to read, will allow us to move some behavior in the LineCoverage class (like aggregating the line values) and to reuse the code when we add the same behaviour or OpenCover.

Reply:
I made a private class because 
- a universal solution should live in the `Coverage` class, and I don't want to come up with this in this PR
- the current solution is tied to the VS coverage XML (e.g. the path to the file lives in a separate section from the actual coverage information)